File: common/src/main/java/org/bread_experts_group/breadlib/task/render/LevelRenderTask.java
package org.bread_experts_group.breadlib.task.render;

import com.mojang.blaze3d.vertex.PoseStack;
import net.minecraft.client.Camera;
import net.minecraft.client.DeltaTracker;
import net.minecraft.client.renderer.LevelRenderer;
import net.minecraft.client.renderer.culling.Frustum;
import org.bread_experts_group.breadlib.task.Task;
import org.jetbrains.annotations.Nullable;
import org.joml.Matrix4f;

public class LevelRenderTask extends Task {
	public final RenderLevelStage stage;
	public final LevelRenderer levelRenderer;
	public final PoseStack poseStack;
	public final Matrix4f projectionMatrix;
	public final DeltaTracker partialTick;
	public final Camera camera;
	public final Frustum frustum;

	public LevelRenderTask(
			RenderLevelStage stage, LevelRenderer levelRenderer, @Nullable PoseStack poseStack,
			Matrix4f projectionMatrix, DeltaTracker partialTick, Camera camera, Frustum frustum
	) {
		super();
		this.stage = stage;
		this.levelRenderer = levelRenderer;
		this.poseStack = poseStack != null ? poseStack : new PoseStack();
		this.projectionMatrix = projectionMatrix;
		this.partialTick = partialTick;
		this.camera = camera;
		this.frustum = frustum;
	}
}

File: common/src/main/kotlin/org/bread_experts_group/breadlib/task/render/LevelRenderTask.kt
package org.bread_experts_group.breadlib.task.render

import com.mojang.blaze3d.vertex.PoseStack
import net.minecraft.client.Camera
import net.minecraft.client.DeltaTracker
import net.minecraft.client.renderer.LevelRenderer
import net.minecraft.client.renderer.culling.Frustum
import org.bread_experts_group.breadlib.task.Task
import org.joml.Matrix4f

class LevelRenderTask(
	val stage: RenderLevelStage, val levelRenderer: LevelRenderer, poseStack: PoseStack?,
	val projectionMatrix: Matrix4f, val partialTick: DeltaTracker, val camera: Camera, val frustum: Frustum
) : Task() {
	val poseStack: PoseStack = poseStack ?: PoseStack()
}